Sources/LeafKit/Data/TemplateDataRepresentable.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+
+/// Capable of being encoded as `TemplateData`.
+public protocol TemplateDataRepresentable {
```

Comment:
I've been trying to move away from this pattern recently because I'm not a fan of how it pollutes the standard library types. Being able to call `"foo".convertToTemplateData()` from everywhere doesn't seem very Swifty.

I think a better pattern is using inits:
```
let data = TemplateData(string: "foo")
```

This method is less generally useful, but that comes with benefits too. Like not needing to use `try` when creating template data from a String.